Add figure not referenced inspection

pschuprikov opened the pull request:
#### Additions

An inspection checking for figure labels that are not anywhere referenced (#875)

_PS. That is my first inspection, so I probably have missed some better way to implement it._

RubenSchellekens commented:
Basically what @PHPirates said! Nice job :) I'll be looking further into it later. I'm about to review all the pull requests.

You might want to add ["figure" to `Magic.Environment`](https://github.com/Ruben-Sten/TeXiFy-IDEA/pull/890/files#diff-7fa16e12eb775f495e5d00cafd52f194R57) as well.

pschuprikov commented:
Thanks! I have added a tooltip and moved the string literal to `Magic`.

There is one problem with the inspection: when I open a Latex file that already has a figure without reference, `getFigureLabels` [returns](https://github.com/Ruben-Sten/TeXiFy-IDEA/pull/890/commits/32b4306e9532a4ebc1ca67f8b6e3f73de7028ac7#diff-7fa16e12eb775f495e5d00cafd52f194R20) empty set, and so the inspection doesn't show up. Do you have any thoughts on why this might be happening?

RubenSchellekens commented:
> There is one problem with the inspection: when I open a Latex file that already has a figure without reference, `getFigureLabels` [returns](https://github.com/Ruben-Sten/TeXiFy-IDEA/pull/890/commits/32b4306e9532a4ebc1ca67f8b6e3f73de7028ac7#diff-7fa16e12eb775f495e5d00cafd52f194R20) empty set, and so the inspection doesn't show up. Do you have any thoughts on why this might be happening?

Probably an index that has not yet been updated. Under the hood you make a call to the following code:
```kotlin
fun PsiFile.findLabels(): Collection<PsiElement> {
    val commands = LatexCommandsIndex.getItems(this)
    val bibtexIds = BibtexIdIndex.getIndexedIds(this)
    val result = ArrayList<PsiElement>(commands)
    result.addAll(bibtexIds)
    return result.findLabels()
}
```
Sometimes (especially new files, or when you just opened them) not all commands/labels are indexed yet. Typing a bit should often resolve this problem. It's a bit annoying, but we still haven't found a way to fix it.

-------------

Awesome that you added the tooltip. Please end the toolip with a period. It might also be a good idea to add some more detailed info (this will show up when you click 'more' in the popup, or look in the settings). A good example of this is the [LatexAvoidEqnarray](https://github.com/Ruben-Sten/TeXiFy-IDEA/blob/master/resources/inspectionDescriptions/LatexAvoidEqnarray.html) inspection. Iirc I've added a small starter in the original issue #875.

--------------------

And lastly, a few nitpicky things:
- Use a continuation indent of 8 instead of 4 (see image).
- Please add a KDoc to Magic.Environments.figures. A magic constant without documentation is still a kind of magic :)

![image](https://user-images.githubusercontent.com/17410729/57489667-03369080-72b7-11e9-9b39-e4c14385aa81.png)

----------

I don't really like pointing out these small relatively minor issues... But I still do, just for quality control :) Thanks for all the effort! It's very much appreciated <3 Do you think the `good-first-issue` tag helps?

pschuprikov commented:
Thanks for a thorough answer! I have applied the suggested changes except for a more detailed inspection description. I am not sure what exactly should go there.

RubenSchellekens commented:
From #875, roughly worded:
> when not referencing a figure you have a random floating image somewhere in your article.

I do not know how to put it into words properly though. But my rationale: Figures that do not get referenced are just 'wandering' inside your document: it is not clear why the figure is there. Figures must be supporting the message you're conveying. If they aren't referenced, there is no logical tie-in with your body and could as well be left out.

If you could rephrase the above maybe? Otherwise I could keep it in the back of my mind and come back to you once I've found something.

PHPirates commented:
@pschuprikov Please don't force push, no need to and it causes very inconvenient merge conflicts for everyone pulling your branch

pschuprikov commented:
Sorry, I won't do that again)
